Bottom: 220121d634269cc5dec0c0d7bc1545acc1e3b8a9
Top:    dfec70d648229122a223c74fd6c162bb7057df14
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-07-11 11:54:39 -0700

librdmacm: Validate source address protocol family in rdma_resolve_addr

If a source address is provided but its protocol family is not recognized,
returns an error.

Signed-off-by: Yann Droneaud <ydroneaud@opteya.com>
Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/src/cma.c b/src/cma.c
index f688c4c..86dd787 100755
--- a/src/cma.c
+++ b/src/cma.c
@@ -811,6 +811,9 @@ int rdma_resolve_addr(struct rdma_cm_id *id, struct sockaddr *src_addr,
 		return ERR(EINVAL);
 
 	src_len = ucma_addrlen(src_addr);
+	if (src_addr && !src_len)
+		return ERR(EINVAL);
+
 	if (af_ib_support)
 		return rdma_resolve_addr2(id, src_addr, src_len, dst_addr,
 					  dst_len, timeout_ms);
